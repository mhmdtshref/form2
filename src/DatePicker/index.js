import React, { Component } from 'react'

class DatePicker extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: (new Date()).toISOString().split('T')[0],
    };
  }

  onChange = ({ target }) => {
    this.setState({ value: target.value }, () => {
      this.props.formstate(this.props.name, this.state.value);
      if(this.props.onChange) this.props.onChange();
    });
  };

  generateProps = (props) => {
    let newProps = {...props};
    delete newProps.onChange;
    delete newProps.formstate;
    delete newProps.value;
    return newProps;
  };

  componentDidMount() {
      this.props.formstate(this.props.name, this.state.value);
  }

  render() {
    return <input type='date' value={this.state.value} onChange={this.onChange} {...this.generateProps(this.props)} />;
  }
}

export default DatePicker;
