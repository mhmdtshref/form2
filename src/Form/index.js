import React, { Component } from 'react'

class Form extends Component {
  constructor(props) {
    super(props)
    this.state = {
      values: {}
    }
  }

  setStateValue = (name, value) => {
    this.setState((prevState) => ({...prevState.values, [name]: value}), () => {
      console.log("STATE CHANGED:::", this.state);
    })
  };

  setEventToState = ({ target }) => {
    this.setStateValue([target.name], target.value)
  };

  renderTextField = (field, index) => {
    field.key = field.key ? field.key : index;
    if (field.onChange) {
      return <input
        type='text'
        name={field.name}
        onChange={(event) => { field.onChange(event); this.setEventToState(event) }}
        {...field}
      />
    } else {
      return <input
        type='text'
        name={field.name}
        onChange={this.setEventToState}
        {...field}
      />
    }
  };

  renderFields = (fields) => {
    return fields.map((field, index) => {
      switch (field.type) {
        case 'text':
          return this.renderTextField(field, index)
      }
    })
  };


  renderChildren = () => {
    const { setStateValue } = this;
    return React.Children.toArray(this.props.children).map((child, key) => React.cloneElement(child, { formstate: setStateValue, key }));
  };

  render() {
    return <form>
      { this.renderChildren() }
    </form>
  }
}

export default Form
