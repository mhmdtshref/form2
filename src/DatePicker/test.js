import React from 'react';
import renderer from 'react-test-renderer';
import DatePicker from "./index";
import Form from "../Form";
import {shallow} from "enzyme/build";

describe('Password Field Test', () => {

  it('is truthy', () => {
    expect(DatePicker).toBeTruthy()
  });

  it('Test natural date picker field', () => {
    const datePicker = shallow(
        <Form>
            <DatePicker name="birthday" value="1996-07-26T01:15" />
        </Form>
    );
    expect(datePicker).toMatchSnapshot();
  });

  it('Test date picker field with default value property', () => {
    const datePciker = shallow(
        <Form>
            <DatePicker name="birthday" value="1996-07-26T01:15" />
        </Form>
    );
    expect(datePciker).toMatchSnapshot();
  });

  it('Test date picker field with onChange function', () => {
    const onChangeHandler = () => {
      // eslint-disable-next-line no-console
      console.log("Test console log!");
    };
    const datePicker = shallow(
        <Form>
            <DatePicker name="birthday" onChange={onChangeHandler} value="1996-07-26T01:15" />
        </Form>
    );
    expect(datePicker).toMatchSnapshot();
  });
});
