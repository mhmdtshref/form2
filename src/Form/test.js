import React from 'react';
import renderer from 'react-test-renderer';
import Form from './index';
import TextField from "../TextField";
import PasswordField from "../PasswordField";
import DatePicker from '../DatePicker';

describe('Form Test', () => {

  it('is truthy', () => {
    expect(Form).toBeTruthy()
  });

  it('Form with one field in it', () => {
    const form = renderer.create(<Form>
      <TextField name="username" />
    </Form>).toJSON();
    expect(form).toMatchSnapshot();
  });

  it('Form with multiple fields in it', () => {
    const form = renderer.create(<Form>
      <TextField name="username" placeholder="Username"/>
      <PasswordField name="password" placeholder="email" />
      <DatePicker name="birthday" />
    </Form>).toJSON();
    expect(form).toMatchSnapshot();
  });
});
