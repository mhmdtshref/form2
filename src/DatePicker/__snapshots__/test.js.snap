// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password Field Test Test date picker field with default value property 1`] = `
<input
  name="birthday"
  onChange={[Function]}
  type="date"
  value="1996-07-26"
/>
`;

exports[`Password Field Test Test date picker field with onChange function 1`] = `
<input
  name="birthday"
  onChange={[Function]}
  type="date"
  value=""
/>
`;

exports[`Password Field Test Test natural date picker field 1`] = `
<input
  name="birthday"
  onChange={[Function]}
  type="date"
  value=""
/>
`;
